yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -2233,65 +2151,119 @@ func (a *SqlAnonymizer) anonymizeColumnDefNode(cd *pg_query.ColumnDef) (err erro
 			continue
 		}
 
-		// Handle A_Const nodes (literal values)
-		aConst := constraint.RawExpr.GetAConst()
-		if aConst == nil || aConst.Val == nil {
-			continue
+		// Handle constant values in default values using the consolidated function
+		if err := a.anonymizeConstantValue(constraint.RawExpr, CONST_KIND_PREFIX); err != nil {
+			return fmt.Errorf("anon coldef default value: %w", err)
 		}
+	}
 
-		// Handle string literals in default values
-		if sval := aConst.GetSval(); sval != nil {
-			sval.Sval, err = a.registry.GetHash(CONST_KIND_PREFIX, sval.Sval)
-			if err != nil {
-				return fmt.Errorf("anon coldef default string value: %w", err)
-			}
-			continue
+	return nil
+}
+
+// anonymizeConstantValue handles anonymization of constant values in SQL statements
+// This function consolidates the logic for handling different types of constant values
+// including TypeName, A_Const (string, int, float, bool), String_, and raw values
+func (a *SqlAnonymizer) anonymizeConstantValue(arg *pg_query.Node, prefix string) error {
+	if arg == nil {
+		return nil
+	}
+
+	// Handle TypeName nodes (e.g., alignment = int4)
+	if typeName := arg.GetTypeName(); typeName != nil {
+		if err := a.anonymizeStringNodes(typeName.Names, prefix); err != nil {
+			return fmt.Errorf("anon constant type value: %w", err)
 		}
+		return nil
+	}
 
-		// Handle integer literals in default values
-		if ival := aConst.GetIval(); ival != nil {
-			// Convert integer to string for hashing
-			intVal := fmt.Sprintf("%d", ival.Ival)
-			hashedVal, err := a.registry.GetHash(CONST_KIND_PREFIX, intVal)
-			if err != nil {
-				return fmt.Errorf("anon coldef default int value: %w", err)
-			}
-			// Replace the integer value node with the string node for anonymized value
-			aConst.Val = &pg_query.A_Const_Sval{
-				Sval: &pg_query.String{Sval: hashedVal},
-			}
-			continue
+	if aConst := arg.GetAConst(); aConst != nil {
+		return a.anonymizeAConst(aConst, prefix)
+	}
+
+	// Handle String_ nodes (e.g., delimiter = ',')
+	if str := arg.GetString_(); str != nil {
+		hashedVal, err := a.registry.GetHash(prefix, str.Sval)
+		if err != nil {
+			return fmt.Errorf("anon constant string value: %w", err)
 		}
+		str.Sval = hashedVal
+		return nil
+	}
 
-		// Handle float literals in default values
-		if fval := aConst.GetFval(); fval != nil {
-			// Get the float value as string and hash it
-			floatVal := fval.Fval
-			hashedVal, err := a.registry.GetHash(CONST_KIND_PREFIX, floatVal)
-			if err != nil {
-				return fmt.Errorf("anon coldef default float value: %w", err)
-			}
-			// Replace the float value node with the string node for anonymized value
-			aConst.Val = &pg_query.A_Const_Sval{
-				Sval: &pg_query.String{Sval: hashedVal},
-			}
-			continue
+	// Handle Integer nodes (e.g., internallength = 1548)
```

Comment:
Question for later: so in some cases, like these length/delimiter-related parameters, it's a generic value and not user-specific information, we might not need to anonymize, as it might be useful information for us to analyze.
But yeah, it's hard to segragate these two things